pkg/machine/fcos.go

```diff
@@ -271,30 +258,69 @@ func GetFCOSDownload(imageStream string, compression imageCompression) (*FcosDow
 	if !ok {
 		return nil, fmt.Errorf("unable to pull VM image: no targetArch in stream")
 	}
-	artifacts := arch.Artifacts
-	if artifacts == nil {
+	upstreamArtifacts := arch.Artifacts
+	if upstreamArtifacts == nil {
 		return nil, fmt.Errorf("unable to pull VM image: no artifact in stream")
 	}
-	qemu, ok := artifacts[Qemu.String()]
+	upstreamArtifact, ok := upstreamArtifacts[vp.Artifact().String()]
 	if !ok {
-		return nil, fmt.Errorf("unable to pull VM image: no qemu artifact in stream")
+		return nil, fmt.Errorf("unable to pull VM image: no %s artifact in stream", vp.Artifact().String())
 	}
-	formats := qemu.Formats
+	formats := upstreamArtifact.Formats
 	if formats == nil {
 		return nil, fmt.Errorf("unable to pull VM image: no formats in stream")
 	}
-	qcow, ok := formats[qcow.String()]
+	formatType, ok := formats[vp.Format().String()]
 	if !ok {
-		return nil, fmt.Errorf("unable to pull VM image: no qcow2.xz format in stream")
+		return nil, fmt.Errorf("unable to pull VM image: no %s format in stream", vp.Format().String())
 	}
-	disk := qcow.Disk
+	disk := formatType.Disk
 	if disk == nil {
 		return nil, fmt.Errorf("unable to pull VM image: no disk in stream")
 	}
 	return &FcosDownloadInfo{
 		Location:        disk.Location,
-		Release:         qemu.Release,
+		Release:         upstreamArtifact.Release,
 		Sha256Sum:       disk.Sha256,
-		CompressionType: compression.String(),
+		CompressionType: vp.Compression().String(),
 	}, nil
 }
+
+type FCOSStream int64
+
+const (
+	// FCOS streams
+	// Testing FCOS stream
+	Testing FCOSStream = iota
+	// Next FCOS stream
+	Next
```

Comment:
Does this work properly? I thought comments between the iota consts messed things up?

Reply:
sure seems to